lib/Service/ShareService.php

```diff
@@ -72,6 +77,7 @@ public function __construct(
 		$this->viewMapper = $viewMapper;
 		$this->userHelper = $userHelper;
 		$this->groupHelper = $groupHelper;
+		$this->circleHelper = $circleHelper;
```

Comment:
For `CircleHelper` has direct dependencies on `\OCA\Circles` this will probably crash when `circles` is disabled. Perhaps also the reason why integration tests are crashing. 

Either it has to be loaded only when it is ensured that circles is loaded (use `IAppManager`) or change `CircleHelper` accordingly.